services/base-jdbc/src/main/java/org/eclipse/hono/service/base/jdbc/config/JdbcProperties.java

```diff
@@ -56,7 +63,10 @@ public JdbcProperties() {
     public JdbcProperties(final JdbcOptions options) {
         Objects.requireNonNull(options);
         setDriverClass(options.driverClass());
-        options.maximumPoolSize().ifPresent(this::setMaximumPoolSize);
+        options.maximumPoolSize().ifPresentOrElse(this::setMaximumPoolSize, () -> setMaximumPoolSize(MAXIMUM_POOL_SIZE_DEFAULT));
+        options.minimumPoolSize().ifPresentOrElse(this::setMinimumPoolSize, () -> setMinimumPoolSize(MINIMUM_POOL_SIZE_DEFAULT));
+        options.initialPoolSize().ifPresentOrElse(this::setInitialPoolSize, () -> setInitialPoolSize(INITIAL_POOL_SIZE_DEFAULT));
+        options.maximumIdleTime().ifPresentOrElse(this::setMaximumIdleTime, () -> setMaximumIdleTime(MAXIMUM_IDLE_TIME_DEFAULT));
```

Comment:
we usually set the default values in the corresponding `*Options` class methods using the `@WithDefault` annotation. For an example, you might want to refer to https://github.com/eclipse-hono/hono/blob/master/core/src/main/java/org/eclipse/hono/config/ServerOptions.java

Reply:
I changed it. Thanks for the tip :)